### PR TITLE
feat: index receipts

### PR DESCRIPTION
**Pull request by greged93**
Adds the transactions receipts indexing.

### CODE REVIEW

File: src/types/receipt.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    logs,
+    logsBloom: logsBloom(logs.map(toLog)),
+    status,
+  };
+}
+
+type Log = [address: Uint8Array, topics: Uint8Array[], data: Uint8Array];
+
+/**
+ * @param log A Ethereum json rpc log.
```
Comment (Eikix) [inline src/types/receipt.ts:70]: i think it's not so clearly written idk it's hard to read

File: src/types/receipt.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  // https://github.com/kkrt-labs/kakarot/blob/main/src/kakarot/accounts/eoa/library.cairo
+  const gasUsed = bigIntToHex(BigInt(event.data[event.data.length - 1]));
+  // Status is the second to last piece of data in the transaction_executed event.
+  // https://github.com/kkrt-labs/kakarot/blob/main/src/kakarot/accounts/eoa/library.cairo
+  const status = bigIntToHex(BigInt(event.data[event.data.length - 2]));
+  // If there is no destination, calculate the deployed contract address.
+  const contractAddress = transaction.to === null
+    ? bytesToHex(generateAddress(
+      hexToBytes(transaction.from),
+      hexToBytes(transaction.nonce),
```
Comment (Eikix) [inline src/types/receipt.ts:42]: i think there's a trap in the nonce:
https://github.com/ethereum/execution-specs/blob/559ba54829a5d726ca95fe5701486f7bd303bf7f/src/ethereum/paris/utils/message.py#L79

the nonce is already incremented no?

**Review by Eikix — COMMENTED** [review]
first review

File: src/main.ts [hunk 3]
```diff
@@ -51,6 +54,18 @@ export default function transform({
       store.push({ collection: "logs", data: { log: ethLog } });
     });
 
+    const ethReceipt = toEthReceipt({
+      transaction: ethTx,
+      logs: ethLogs,
+      event,
+      cumulativeGasUsed,
+    });
+    if (ethReceipt === null) {
```
Comment (Eikix) [inline src/main.ts:63]: under which circumstances is this null?

i think we can also add it on ethTx / logs -> when can they be null?

File: src/types/receipt.ts (hunk 1 above)
Comment (greged93) [reply src/types/receipt.ts:70]: I'm not sure what you mean, the comment is difficult to read?

File: src/main.ts [hunk 4]
```diff
@@ -35,22 +36,45 @@ export default function transform({
...
+    // 1. The event is part of the defined ignored events (see IGNORED_KEYS).
+    // 2. The event has an invalid number of keys.
     const ethLogs = receipt.events.map((e) => {
       return toEthLog({ transaction: ethTx, event: e });
     }).filter((e) => e !== null) as JsonRpcLog[];
     ethLogs.forEach((ethLog) => {
       store.push({ collection: "logs", data: { log: ethLog } });
     });
 
+    const ethReceipt = toEthReceipt({
```
Comment (Eikix) [inline src/main.ts:69]: the eth receipt cannot be null?

File: src/types/receipt.ts (hunk 1 above)
Comment (Eikix) [reply src/types/receipt.ts:70]: yes

Comment (Eikix) [reply src/types/receipt.ts:70]:
```suggestion
 * @param log - JSON RPC formatted Ethereum Log.
```

File: src/main.ts (hunk 4 above)
Comment (greged93) [reply src/main.ts:69]: No, if the transaction and logs are ok, the receipt will be ok :)

**Review by Eikix — APPROVED** [review]
lgtm

**Pull request merged**